### PR TITLE
Local configuration negation

### PR DESCRIPTION
**Pull request by Chris--S**
Allows (non-php) configuration values set earlier in the config cascade to be reversed.  E.g. to counter a default configuration value.  

For key/value config files (e.g. acronyms), include a key with no value
```
# to set a value
HTML      hypertext markup language
# to unset a value
HTML
```

For value only config files (e.g. scheme, stopwords), prefix the value with an exclamation mark, '!'.
```
# to set a value
gopher
# to unset a value
!gopher
```

### CODE REVIEW

**Comment by michitux** [comment]
So basically this means that you cannot set any of the configuration values to 0 anymore. I'm not sure if this is useful/if we should keep this possibility. Apart from that: :+1:

**Comment by Chris--S** [comment]
patch pushed to address @michitux concern.

also to make negation using '!' resistant to white space differences.

File: inc/confutils.php [hunk 1]
```diff
@@ -196,7 +208,7 @@ function confToHash($file,$lower=false) {
  * @param  array    $params   optional additional params to pass to the callback
  * @return array    configuration values
  */
-function retrieveConfig($type,$fn,$params=null) {
+function retrieveConfig($type,$fn,$params=null,$combine='array_merge') {
```
Comment (splitbrain) [inline inc/confutils.php]: the doc block should be adjusted for the new paramter

**Comment by splitbrain** [comment]
This means we can no longer have entities starting with a ```!```, right? This is not a theoretical problem. Our current entity.conf on dokuwiki.org has these lines currently:

```
!cr!  <div style="clear: right; float: right;"></div>
!cl!  <div style="clear: left; float: left;"></div>
!cb!  <div class="clearer"></div>
```

**Comment by Chris--S** [comment]
No.  Entities are key/value conf., so are cleared by putting no value
for the key.  Key is unchanged.

Only scheme and stopwords are single value, so needed a different
method, hence using a '!' prefix.

- Chris

**Comment by splitbrain** [comment]
ah. good to merge then :-)

**Comment by Klap-in** [comment]
Needs still an update of the docs

**Pull request merged**